<launch>
    <arg name="ns" default="os1_node"/>

    <arg name="lidar_frame" default="lidar"/>
    <arg name="imu_frame" default="imu"/>

    <arg name="image_width" default="1024"/>
    <arg name="use_intensity" default="True"/>
    <arg name="full_sweep" default="True"/>

    <arg name="debug" default="false"/>
    <env if="$(arg debug)" name="ROSCONSOLE_CONFIG_FILE" value="$(find ouster_ros)/launch/debug.conf"/>

    <node pkg="ouster_ros" type="os1_decoder" name="os1_decoder" output="screen" ns="$(arg ns)">
        <param name="image_width" type="int" value="$(arg image_width)"/>
        <param name="use_intensity" type="bool" value="$(arg use_intensity)"/>
        <param name="full_sweep" type="bool" value="$(arg full_sweep)"/>
        <param name="lidar_frame" type="string" value="$(arg lidar_frame)"/>
        <param name="imu_frame" type="string" value="$(arg imu_frame)"/>

        <remap from="~os1_config" to="os1_config"/>
        <remap from="~lidar_packets" to="lidar_packets"/>
        <remap from="~imu_packets" to="imu_packets"/>

        <remap from="~imu" to="imu"/>
        <remap from="~cloud" to="cloud"/>
        <remap from="~image" to="image"/>
        <remap from="~camera_info" to="camera_info"/>
    </node>

</launch>
